sdk/core/core-lro/src/lroPoller/stateMachine.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { processAzureAsyncOperationResult } from "./azureAsyncPolling";
+import { isBodyPollingDone, processBodyPollingOperationResult } from "./bodyPolling";
+import { processLocationPollingOperationResult } from "./locationPolling";
+import {
+  FinalStateVia,
+  LRO,
+  LROConfig,
+  LROStatus,
+  PollerConfig,
+  RawResponse,
+  ResumablePollOperationState
+} from "./models";
+import { processPassthroughOperationResult } from "./passthrough";
+import { getPollingURL, inferLROMode } from "./requestUtils";
+
+/**
+ * creates a stepping function that maps an LRO state to another.
+ */
+export function createGetLROStatusFromResponse<TResult>(
+  lroPrimitives: LRO<TResult>,
+  config: LROConfig,
+  finalStateVia?: FinalStateVia
+): (rawResponse: RawResponse, flatResponse: TResult) => LROStatus<TResult> {
+  switch (config.mode) {
+    case "AzureAsync": {
+      return processAzureAsyncOperationResult(
+        lroPrimitives,
+        config.resourceLocation,
+        finalStateVia
+      );
+    }
+    case "Location": {
+      return processLocationPollingOperationResult;
+    }
+    case "Body": {
+      return processBodyPollingOperationResult;
+    }
+    default: {
+      return processPassthroughOperationResult;
+    }
+  }
+}
+
+/**
+ * Creates a polling operation that returns a LRO state.
+ */
+export function createPollForLROStatus<TResult>(
+  lroPrimitives: LRO<TResult>,
+  config: LROConfig
+): (pollingURL: string, pollerConfig: PollerConfig) => Promise<LROStatus<TResult>> {
+  return async (path: string, pollerConfig: PollerConfig): Promise<LROStatus<TResult>> => {
+    const response = await lroPrimitives.sendPollRequest(config, path);
+    const retryAfter: string | undefined = response.rawResponse.headers["Retry-After"];
```

Comment:
this is one of the places where I'm wondering if we can assume all services properly capitalize `Retry-After`

Reply:
I am not familiar with this tbh, should I also expect `retry-after`? is there any other name the service could use for this?

Reply:
It's possible our services always return `Retry-After` and this will never be a problem. I think it's just one of those potential issues, where it is perfectly valid for the server to return `retry-after` or `rEtRy-AfTeR` and it would still be valid.

If you're using toJson() of our HttpHeaders classes then I think we normalize to all lowercase

Reply:
I see, this is a bug then. I will fix it.

Reply:
used `retry-after` instead and added a comment on the `headers` field that it is already normalized.